### PR TITLE
[WIP]about create histogram

### PR DESCRIPTION
**Pull request by wolfstudy**
not complete about create histogram

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tikv-client-lib-java/67)

### CODE REVIEW

File: src/main/java/com/pingcap/tikv/statistics/Histogram.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    public Bucket[] buckets;
+
+    private static TiConfiguration conf = TiConfiguration.createDefault(ImmutableList.of("127.0.0.1:" + 2379));
+    private static TiCluster cluster = TiCluster.getCluster(conf);
+    private static Snapshot snapshot = cluster.createSnapshot();
+
+    // histogramFromStorage from the storage to histogram.
+    public void histogramFromStorage(long tableID,long isIndex,long colID){
+        Catalog cat = cluster.getCatalog();
+        TiDBInfo db = cat.getDatabase("mysql");
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/statistics/Histogram.java]: please do not do hard code. Instead, you can do the following:

private final String DBNAME = "mysql";

File: src/main/java/com/pingcap/tikv/statistics/Histogram.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        );
+
+        ScanBuilder scanBuilder = new ScanBuilder();
+        ScanBuilder.ScanPlan scanPlan = scanBuilder.buildScan(firstAnd, index, table);
+
+        TiSelectRequest selReq = new TiSelectRequest();
+        selReq.addRanges(scanPlan.getKeyRanges())
+                .setTableInfo(table)
+                .setIndexInfo(index)
+                .addField(TiColumnRef.create("table_id", table))
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/statistics/Histogram.java]: this is hard code too, please address this like I said in `DBNAME`.

File: src/main/java/com/pingcap/tikv/statistics/Histogram.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                .addField(TiColumnRef.create("is_index", table))
+                .addField(TiColumnRef.create("hist_id", table))
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/statistics/Histogram.java]: When double read PR merged, you need update `addField` to `addRequiredColumn`.

File: src/main/java/com/pingcap/tikv/statistics/Histogram.java [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+                if(isIndex == 1) {
+                    Bucket.lowerBound = Comparables.wrap(row.getLong(3));
+                }else {
+                    Bucket.lowerBound = (Comparable<ByteString>)
+                            row.get(row.getInteger(3), DataTypeFactory.of(Types.TYPE_BLOB));
+                    Bucket.upperBound = (Comparable<ByteString>)
+                            row.get(row.getInteger(4), DataTypeFactory.of(Types.TYPE_BLOB));
+                }
+                for (int i = 0; i < row.fieldCount(); i++) {
+                    Object val = row.get(i, schemaInfer.getType(i));
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/statistics/Histogram.java]: why do you do this? the value of row has been extracted already. In addition, I did not see any where that you are creating a histogram. As the method name suggests, this method will retrieve a histogram from `TiKV`. But you did not save such information.

File: src/main/java/com/pingcap/tikv/statistics/Histogram.java (hunk 3 above)
Comment (wolfstudy) [reply src/main/java/com/pingcap/tikv/statistics/Histogram.java]: ok  when double read PR merged ,i will uodate ``` addField```  to ```addRequiredColumn```.I will pay close attention to the PR at any time

**Pull request closed without merging**